### PR TITLE
Homework mock

### PR DESCRIPTION
**Pull request by mariuszlisowski**

### CODE REVIEW

File: test/PizzeriaTest.cpp [hunk 1]
```diff
@@ -52,16 +56,52 @@ TEST_F(PizzeriaTest, completeOrderWithStubPizza)
...
 {   
     // Given
-    PizzaMock* mock = new PizzaMock{};
+    NaggyMock<PizzaMock>* mock = new NaggyMock<PizzaMock>{};
     Pizzas pizzas = {mock};
+    EXPECT_CALL(*mock, getName()).WillOnce(Return("naggy_mock"));
     EXPECT_CALL(*mock, getPrice()).WillOnce(Return(40.0));
-    
+    EXPECT_CALL(*mock, getBakingTime).Times(1);
+    EXPECT_CALL(mockTimer, sleepFor).Times(1);
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:64]:
```suggestion
    constexpr minutes mockBakingTime = minutes(10);
    EXPECT_CALL(*mock, getBakingTime).WillOnce(Return(mockBakingTime);
    EXPECT_CALL(mockTimer, sleepFor(mockBakingTime)).Times(1);
```

File: test/PizzeriaTest.cpp [hunk 2]
```diff
@@ -52,16 +56,52 @@ TEST_F(PizzeriaTest, completeOrderWithStubPizza)
...
+
+    EXPECT_CALL(*strictMockPizza, getName()).WillOnce(Return("scrict_mock"));
+    EXPECT_CALL(*strictMockPizza, getBakingTime()).WillOnce(Return(minutes(1)));
+    EXPECT_CALL(*strictMockPizza, getPrice()).WillOnce(Return(30.0));
+
+    EXPECT_CALL(*niceMockPizza, getName()).WillOnce(Return("nice_mock"));
+    EXPECT_CALL(*niceMockPizza, getBakingTime()).WillOnce(Return(minutes(1)));
+    EXPECT_CALL(*niceMockPizza, getPrice()).WillOnce(Return(35.0));
+
+    EXPECT_CALL(mockTimer, sleepFor).Times(3);
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:93]: you should expect 3 calls, but each with different parameter - baking time of each pizza.

File: test/PizzeriaTest.cpp [hunk 3]
```diff
@@ -52,16 +56,52 @@ TEST_F(PizzeriaTest, completeOrderWithStubPizza)
...
+
+    Pizzas pizzas = {stubPizza, strictMockPizza, niceMockPizza};
+
+    EXPECT_CALL(*strictMockPizza, getName()).WillOnce(Return("scrict_mock"));
+    EXPECT_CALL(*strictMockPizza, getBakingTime()).WillOnce(Return(minutes(1)));
+    EXPECT_CALL(*strictMockPizza, getPrice()).WillOnce(Return(30.0));
+
+    EXPECT_CALL(*niceMockPizza, getName()).WillOnce(Return("nice_mock"));
+    EXPECT_CALL(*niceMockPizza, getBakingTime()).WillOnce(Return(minutes(1)));
+    EXPECT_CALL(*niceMockPizza, getPrice()).WillOnce(Return(35.0));
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:91]: Magic values. Use constexpr.

File: test/PizzeriaTest.cpp [hunk 4]
```diff
@@ -52,16 +56,52 @@ TEST_F(PizzeriaTest, completeOrderWithStubPizza)
...
+    EXPECT_CALL(mockTimer, sleepFor).Times(3);
+
+    // When
+    auto orderId = pizzeria.makeOrder(pizzas);
+    pizzeria.bakePizzas(orderId);
+    pizzeria.completeOrder(orderId);
+    auto price = pizzeria.calculatePrice(orderId);
+
+    // Then
+    ASSERT_EQ(65, price);
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:102]: Magic value

**Review by ziobron — CHANGES_REQUESTED** [review]

**Review by WBigaj — COMMENTED** [review]
I have checked your code and have nothing to add.

File: test/PizzeriaTest.cpp (hunk 1 above)
Comment (mariuszlisowski) [reply test/PizzeriaTest.cpp:64]: Replaced as suggested

File: test/PizzeriaTest.cpp (hunk 2 above)
Comment (mariuszlisowski) [reply test/PizzeriaTest.cpp:93]: Indeed. Add three calls, each with different parameter.

File: test/PizzeriaTest.cpp (hunk 3 above)
Comment (mariuszlisowski) [reply test/PizzeriaTest.cpp:91]: Constexpr used.

File: test/PizzeriaTest.cpp (hunk 4 above)
Comment (mariuszlisowski) [reply test/PizzeriaTest.cpp:102]: Constexpr used.

**Review by niemarcin — APPROVED** [review]
Looks good

File: test/PizzeriaTest.cpp [hunk 5]
```diff
@@ -10,8 +10,15 @@
...
 
-struct PizzeriaTest : public ::testing::Test
-{
+constexpr minutes mockBakingTime = minutes(10);
+constexpr minutes strictBakingTime = minutes(12);
+constexpr minutes niceBakingTime = minutes(16);
+
+constexpr double mockPizzaPrice = 40;
+constexpr double strictMockPizzaPrice = 30;
+constexpr double niceMockPizzaPrice = 35;
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:19]:
```suggestion
constexpr double mockPizzaPrice = 40.0;
constexpr double strictMockPizzaPrice = 30.0;
constexpr double niceMockPizzaPrice = 35.0;
```

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
{ "requirement_ids" : [1119, 1119, 1119, 3524, 3525, 3526] }
3 * 3 XP za punktualność (1119)
10 XP za TimeMock w Pizzas (3524)
15 XP za nowy przypadek testowy w Pizzas (3525)
5 XP za workflow w Pizzas (3526)

**Pull request closed without merging**